pkg/query-service/app/logs/v3/enrich_query.go

```diff
@@ -149,3 +151,69 @@ func enrichFieldWithMetadata(field v3.AttributeKey, fields map[string]v3.Attribu
 	field.DataType = v3.AttributeKeyDataTypeString
 	return field
 }
+
+func jsonFilterEnrich(filter v3.FilterItem) v3.FilterItem {
+	// check if it is a json request
+	if !strings.HasPrefix(filter.Key.Key, "body.") {
+		return filter
+	}
+
+	// check if the value is a int, float, string, bool
+	valueType := ""
+	switch filter.Value.(type) {
+	case uint8, uint16, uint32, uint64, int, int8, int16, int32, int64:
+		valueType = "int64"
+	case float32, float64:
+		valueType = "float64"
+	case string:
+		valueType, filter.Value = parseStrValue(filter.Value.(string), filter.Operator)
+	case bool:
+		valueType = "bool"
+	}
+
+	// check if it is array
+	if strings.HasSuffix(filter.Key.Key, "[*]") {
+		valueType = fmt.Sprintf("array(%s)", valueType)
+	}
+
+	filter.Key.DataType = v3.AttributeKeyDataType(valueType)
+	filter.Key.IsJSON = true
+	return filter
+}
+
+func parseStrValue(valueStr string, operator v3.FilterOperator) (string, interface{}) {
+
+	valueType := "string"
+
+	// for the following operators it will always be string
+	if operator == v3.FilterOperatorContains || operator == v3.FilterOperatorNotContains ||
+		operator == v3.FilterOperatorRegex || operator == v3.FilterOperatorNotRegex ||
+		operator == v3.FilterOperatorLike || operator == v3.FilterOperatorNotLike {
+		return valueType, valueStr
+	}
+
+	var err error
+	var value interface{}
+
+	// try to convert it to bool/int/float
+	value, err = strconv.ParseBool(valueStr)
+	if err != nil {
+		// try parsing it as int
+		value, err = strconv.ParseInt(valueStr, 10, 64)
+		// try parsing it as float
+		if err != nil {
+			value, err = strconv.ParseFloat(valueStr, 64)
+			if err != nil {
+				value = valueStr
+			} else {
+				valueType = "float64"
+			}
+		} else {
+			valueType = "int64"
+		}
+	} else {
+		valueType = "bool"
+	}
```

Comment:
The deep nested conditions was hard to read. Might be better to refactor something like this.
```golang
if parsedValue, err := strconv.ParseBool(valueStr); err == nil {
    value = parsedValue
    valueType = "bool"
} else if parsedValue, err := strconv.ParseInt(valueStr, 10, 64); err == nil {
    value = parsedValue
    valueType = "int64"
} else if parsedValue, err := strconv.ParseFloat(valueStr, 64); err == nil {
    value = parsedValue
    valueType = "float64"
} else {
    value = valueStr
    valueType = "string"
}
```

Reply:
Done